### PR TITLE
RFC: fix CI, add PyPI publish action, poetry 1.6.1

### PR DESCRIPTION
**Pull request by leo-mansini**
Adds github action to publish to pypi. Also tries with testpypi on every push (not just a tag release).

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: test
+on: push
+
+jobs:
+    pypi-publish:
+        if: startsWith(github.ref, 'refs/tags/')
```
Comment (MartinCura) [inline .github/workflows/publish.yml:6]: this makes sure that this runs only when a tag is pushed, right?

Comment (leo-mansini) [reply .github/workflows/publish.yml:6]: Yes, I couldn't find a more robust way to do it, but it's the suggested way in the action documentation and in the [python guide](https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#the-whole-ci-cd-workflow).

Comment (MartinCura) [reply .github/workflows/publish.yml]: great, adjusted it a bit based on the action's recommendations

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: publish
+on: push
+
+env:
+  PYTHON_VERSION: 3.10.10
+  POETRY_VERSION: 1.6.1
+
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
```
Comment (leo-mansini) [inline .github/workflows/publish.yml:10]:
```suggestion
    name: Publish Python 🐍 distribution 📦 to PyPI
```

Just to keep both job names :sparkles: fun :sparkles:

**Review by leo-mansini — COMMENTED** [review]
Can't approve as I opened the PR but consider it approved. 📦 :snake:

**Review by MartinCura — APPROVED** [review]

**Pull request merged**